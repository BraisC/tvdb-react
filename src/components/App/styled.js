import styled from 'styled-components';

const Wrapper = styled.div`
  display: grid;
  min-height: 100vh;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    'header'
    'content'
    'footer';
`;

const ContentWrapper = styled.div`
  grid-area: content;
`;

export const Styled = {
  Wrapper,
  ContentWrapper,
};
