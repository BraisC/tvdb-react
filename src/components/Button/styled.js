import styled from 'styled-components';

const Button = styled.div`
  padding: 0.8rem 3rem;
  background-color: var(--color-red);
  color: var(--color-white);
  font-size: 1.6rem;
  display: flex;
  align-items: center;
`;

export const Styled = {
  Button,
};
