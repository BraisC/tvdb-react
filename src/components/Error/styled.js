import styled from 'styled-components';

const Wrapper = styled.div`
  color: var(--color-text);
  padding: 3rem 6rem;
  display: flex;
  flex-direction: column;
  transition: color 0.2s ease;
  align-items: center;
`;

const PageTitle = styled.h1`
  font-size: 2.5rem;
  font-weight: 100;
  text-transform: uppercase;
  margin-top: 4rem;
`;

const ImageWrapper = styled.picture`
  width: 40%;
`;

const Image = styled.img`
  width: 100%;
`;

export const Styled = {
  Wrapper,
  PageTitle,
  ImageWrapper,
  Image,
};
