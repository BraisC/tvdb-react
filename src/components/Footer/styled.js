import styled from 'styled-components';

const Footer = styled.footer`
  grid-area: footer;
  background: var(--color-primary);
  transition: 0.2s ease;
`;

export const Styled = {
  Footer,
};
