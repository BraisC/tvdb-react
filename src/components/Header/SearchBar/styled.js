import styled from 'styled-components';

const SearchBar = styled.form`
  width: 40rem;
  margin-right: auto;
  display: flex;
  align-items: center;
`;

const SearchInput = styled.input`
  width: 100%;
  background: transparent;
  border: none;
  font-size: 1.5rem;
  color: var(--color-text);
  transition: background-color 0.2s ease, color 0.2s ease;
  padding-left: 5rem;
  font-family: Montserrat;

  &:focus::placeholder {
    opacity: 0;
  }

  &::placeholder {
    opacity: 35%;
    text-transform: uppercase;
    font-family: Montserrat;
    transition: 0.2s ease;
  }
`;

const SearchIcon = styled.label`
  color: var(--color-red);
  margin-right: -3.5rem;
  margin-top: 3px;
  font-size: 2.2rem;
  cursor: pointer;
  z-index: 999;
`;

export const Styled = {
  SearchBar,
  SearchInput,
  SearchIcon,
};
