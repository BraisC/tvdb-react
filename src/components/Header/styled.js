import styled from 'styled-components';

const Header = styled.header`
  display: flex;
  background: var(--color-primary);
  align-items: center;
  transition: background-color 0.2s ease, color 0.2s ease;
  grid-area: header;
  position: relative;
  z-index: 3;
`;

const Logo = styled.div`
  background-color: var(--color-red);
  color: var(--color-white);
  font-size: 3rem;
  font-weight: bold;
  padding: 1.3rem 5rem;
`;

const SubMenu = styled.ul`
  display: grid;
  position: absolute;
  transform: translateY(50%);
  opacity: 0;
  background-color: var(--color-primary);
  transition: all 0.5s ease;
  grid-template-rows: repeat(8, auto);
  grid-auto-flow: column;
  padding: 1.5rem 1rem;
  visibility: hidden;
  pointer-events: none;

  & li a {
    padding: 0.5rem 1rem;
    font-size: 1.4rem;

    display: block;
  }

  &:hover {
    transform: translateY(0);
    opacity: 1;
    visibility: visible;
  }
`;

const SubMenuButton = styled.span`
  cursor: pointer;
`;

const SubMenuWrapper = styled.li`
  &:hover > ${SubMenu} {
    transform: translateY(0);
    opacity: 1;
    visibility: visible;
    pointer-events: auto;
  }
`;

const Menu = styled.ul`
  padding: 0 2rem;
  display: flex;
  margin-right: auto;

  & li {
    list-style: none;
  }

  & a,
  & ${SubMenuButton} {
    color: var(--color-text);
    font-size: 1.8rem;
    text-transform: uppercase;
    text-decoration: none;
    padding: 1rem 3rem;
    transition: 0.2s ease;

    &:hover {
      color: var(--color-red);
    }

    & span {
      padding: 0.2rem 0rem;
    }

    &.active span {
      border-bottom: 2px solid var(--color-red);
    }
  }
`;

const SearchBar = styled.form`
  width: 40rem;
  margin-right: auto;
  display: flex;
  align-items: center;
`;

const SearchInput = styled.input`
  width: 100%;
  background: transparent;
  border: none;
  font-size: 1.5rem;
  color: var(--color-text);
  transition: 0.2s ease;
  padding-left: 5rem;

  &:focus::placeholder {
    opacity: 0;
  }

  &::placeholder {
    opacity: 35%;
    color: var(--color-text);
    text-transform: uppercase;
    font-family: Montserrat;
  }
`;

const SearchIcon = styled.label`
  color: var(--color-red);
  margin-right: -3.5rem;
  font-size: 2.2rem;
  cursor: pointer;
  z-index: 999;
`;

export const Styled = {
  Header,
  Logo,
  Menu,
  SearchBar,
  SearchInput,
  SearchIcon,
  SubMenuButton,
  SubMenu,
  SubMenuWrapper,
};
