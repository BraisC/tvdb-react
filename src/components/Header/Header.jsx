import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';
import { GenresContext } from 'contexts/genresContext';
import { Styled } from './styled';
import { SearchBar } from './SearchBar';
import { ThemeToggler } from './ThemeToggler';

const MenuOptions = [
  { name: 'popular', route: '/' },
  { name: 'top', route: 'top rated' },
  { name: 'on air', route: 'on the air' },
  { name: 'today', route: 'airing today' },
  { name: 'genre' },
];

const SubMenuItem = ({ value }) => (
  <li>
    <NavLink to={`${process.env.PUBLIC_URL}/genre/${value}`} exact>
      <span>{`${value}`}</span>
    </NavLink>
  </li>
);

const SubMenu = ({ value, subvalues }) => (
  <Styled.SubMenuWrapper>
    <Styled.SubMenuButton>{value}</Styled.SubMenuButton>
    <Styled.SubMenu>
      {subvalues ? subvalues?.map((s) => <SubMenuItem key={s.name} value={s.name} />) : null}
    </Styled.SubMenu>
  </Styled.SubMenuWrapper>
);

const MenuItem = ({ value, route }) => (
  <li>
    <NavLink to={`${process.env.PUBLIC_URL}/shows/${route}`.replace('shows//', '')} exact>
      <span>{`${value}`}</span>
    </NavLink>
  </li>
);

const Menu = () => {
  const genres = useContext(GenresContext);

  return (
    <Styled.Menu>
      {MenuOptions.map((val) =>
        val.route ? (
          <MenuItem key={val.name} value={val.name} route={val.route} />
        ) : (
          <SubMenu key={val.name} value={val.name} subvalues={genres} />
        )
      )}
    </Styled.Menu>
  );
};

const Header = () => (
  <Styled.Header>
    <Styled.Logo>TVDB</Styled.Logo>
    <Menu />
    <SearchBar />
    <ThemeToggler />
  </Styled.Header>
);

export default Header;
