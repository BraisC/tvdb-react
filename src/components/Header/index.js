import Header from './Header';

export { default } from './Header.jsx';
