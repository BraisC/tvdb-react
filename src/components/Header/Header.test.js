import React, { useState as useStateMock } from 'react';
import Enzyme, { shallow } from 'enzyme';
import Header, { SubMenu } from './Header';
import { Styled } from './styled';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('<Header/>', () => {
  test('if Header component renders', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  });

  describe('<Submenu/ >', () => {
    let container;
    const setState = jest.fn();

    beforeEach(() => {
      useStateMock.mockImplementation((init) => [init, setState]);
      container = Enzyme.shallow(<SubMenu />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('Hover', () => {
      test('if shows submenu on hover', () => {
        container.find(Styled.SubMenuWrapper).simulate('mouseEnter');
        expect(setState).toHaveBeenCalledWith(true);
      });
      test('if hides submenu on mouseleave', () => {
        container.find(Styled.SubMenuWrapper).simulate('mouseLeave');
        expect(setState).toHaveBeenCalled();
      });
    });
  });
});
