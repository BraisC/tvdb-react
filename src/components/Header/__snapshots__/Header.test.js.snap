// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> if Header component renders 1`] = `ShallowWrapper {}`;
